tuxemon/save.py

```diff
@@ -52,127 +54,172 @@
+	"""
+	Gets a dictionary which represents the state of the session.
 
-def capture_screenshot(client: LocalPygameClient) -> pygame.surface.Surface:
-    """
-    Capture a screenshot.
+	Parameters:
+		session: Game session.
 
-    Parameters:
-        client: Tuxemon client.
+	Returns:
+		Game data to save, must be JSON encodable.
 
-    Returns:
-        Captured image.
+	"""
+	save_data = session.player.get_state(session)
+	screenshot = capture_screenshot(session.client)
+	save_data["screenshot"] = base64.b64encode(pygame.image.tostring(screenshot, "RGB")).decode("utf-8")
+	save_data["screenshot_width"] = screenshot.get_width()
+	save_data["screenshot_height"] = screenshot.get_height()
+	save_data["time"] = datetime.datetime.now().strftime(TIME_FORMAT)
+	save_data["version"] = SAVE_VERSION
+	return save_data
 
-    """
-    screenshot = pygame.Surface(client.screen.get_size())
-    world = client.get_state_by_name(WorldState)
-    assert world
-    world.draw(screenshot)
-    return screenshot
+def open_save_file(save_path: str) -> Optional[Dict[str, Any]]:
 
+	if config.compress_save:
+		try:
+			DATA = compress_json.load(
+				save_path
+			)
+			return DATA
+		except Exception as e:
+			logger.info(e)
+			return None
+	else:
+		try:
+			with open(save_path) as save_file:
+				try:
+					return json.load(save_file)
+				except ValueError as e:
+					logger.error("cannot decode JSON: %s", save_path)
+
+				try:
+					return cbor.load(save_file)
+				except ValueError as e:
+					logger.error("cannot decode save CBOR: %s", save_path)
+
+				return {}
+
+		except OSError as e:
+			logger.info(e)
+			return None
 
 def save(
-    save_data: Mapping[str, Any],
-    slot: int,
+	save_data: Mapping[str, Any],
+	slot: int,
 ) -> None:
-    """
-    Saves the current game state to a file using CBOR or JSON.
-
-    Parameters:
-        save_data: The data to save.
-        slot: The save slot to save the data to.
-
-    """
-    # Save a screenshot of the current frame
-    save_path = prepare.SAVE_PATH + str(slot) + ".save"
-    if prepare.SAVE_METHOD == "CBOR":
-        text = cbor.dumps(save_data)
-    else:
-        text = json.dumps(save_data, indent=4, separators=(",", ": "))
-    with open(save_path, "w") as f:
-        logger.info("Saving data to save file: " + save_path)
-        # Don't dump straight to the file: if we crash it would corrupt
-        # the save_data
-        f.write(text)
+	"""
+	Saves the current game state to a file using gzip compressed JSON.
+
+	Parameters:
+		save_data: The data to save.
+		slot: The save slot to save the data to.
+
+	"""
+	# Save a screenshot of the current frame
+	# if/else paradise
+	if config.compress_save:
+		if config.csave_lzma:
+			save_path = prepare.SAVE_PATH + str(slot) + ".csave.lza"
+			text = compress_json.dump(
+			save_data,
+			save_path,
+			compression_kwargs = {},
+			json_kwargs = { "indent": 4, 'separators': (",", ": ") }
+		)
+		else:
```

Comment:
the only thing changing here is the filename and compression kwargs.  this should be refactored to remove the redundant code in these if/else blocks